src/Voron/Impl/Journal/JournalReader.cs

```diff
@@ -121,16 +126,22 @@ public bool ReadOneTransactionToDataFile(StorageEnvironmentOptions options)
                 var numberOfPagesOnDestination = GetNumberOfPagesFor(pageInfoPtr[i].Size);
                 _dataPager.EnsureContinuous(pageInfoPtr[i].PageNumber, numberOfPagesOnDestination);
                 _dataPager.EnsureMapped(this, pageInfoPtr[i].PageNumber, numberOfPagesOnDestination);
-                var pagePtr = _dataPager.AcquirePagePointer(this, pageInfoPtr[i].PageNumber);
 
-                var diffPageNumber = *(long*)(outputPage + totalRead);
-                if (pageInfoPtr[i].PageNumber != diffPageNumber)
-                    throw new InvalidDataException($"Expected a diff for page {pageInfoPtr[i].PageNumber} but got one for {diffPageNumber}");
+                // There is a chance the page was not synced to the data file before we crashed.
+                // If encryption is enabled, when recovering, we cannot decrypt an empty (zeroed) page
+                var pagePtr = options.EncryptionEnabled 
+                    ? _dataPager.AcquirePagePointerForNewPage(this, pageInfoPtr[i].PageNumber, numberOfPagesOnDestination) 
```

Comment:
Probably want to always have a new page here, and need to explain that we also do that for verification purposes, because we are going to overwrite the page anyway, so we don't care about its current contents